### PR TITLE
add cppcheck action and fix issues

### PR DESCRIPTION
**Pull request by mlohvynenko**

### CODE REVIEW

File: include/aos/common/tools/memory.hpp [hunk 1]
```diff
@@ -235,12 +235,14 @@ class SharedPtr : public SmartPtr<T> {
...
     }
 
+    // cppcheck-suppress-begin noExplicitConstructor
     /**
      * Creates shared pointer from another derived class shared pointer.
      *
      * @param ptr pointer to create from.
      */
     template <typename P>
+    // cppcheck-suppress noExplicitConstructor
```
Comment (mykola-kobets-epam) [inline include/aos/common/tools/memory.hpp:245]: may be it would be better to make the constructor explicit?

**Review by mykola-kobets-epam — APPROVED** [review]

**Review by MykolaSolyanko — APPROVED** [review]

File: include/aos/common/tools/memory.hpp (hunk 1 above)
Comment (mlohvynenko) [reply include/aos/common/tools/memory.hpp:367]: llvm STL implementation doesn't mark template copy constructor as explicit.

https://github.com/llvm/llvm-project/blob/main/libcxx/include/__memory/shared_ptr.h#L571

Agreed with Oleksandr to follow the same  approach as llvm

File: include/aos/common/tools/time.hpp [hunk 2]
```diff
@@ -239,10 +239,10 @@ class Time {
      */
     friend Log& operator<<(Log& log, const Time& obj)
     {
-        static constexpr auto cTimeStrLen = 32;
+        static constexpr auto cUTCTimeStrLen = 32;
```
Comment (al1img) [inline include/aos/common/tools/time.hpp:242]: Why it is renamed?

Comment (mlohvynenko) [reply include/aos/common/tools/time.hpp:242]: This constant shadows cTimeStrLen declared in the aos/comon/tools/time.hpp

Comment (al1img) [reply include/aos/common/tools/time.hpp:242]: Then cTimeStrLen should be used instead, no need to add new one. They look the same.

Comment (mlohvynenko) [reply include/aos/common/tools/time.hpp:242]: cTimeStrLen == 68
author of this code specified that string size will be 32.

So I see 2 options:
1. simply use cTimeStrLen global constant (68 bytes)
2. add a new constant (like AOS_CONFIG_TIME_UTS_STR_LEN 32) and use it

Comment (mlohvynenko) [reply include/aos/common/tools/time.hpp]: reused global constant.

**Review by al1img — APPROVED** [review]

**Pull request merged**